tests/migrations/test_operations.py

```diff
@@ -4366,6 +4366,170 @@ def test_remove_constraint(self):
             {"model_name": "Pony", "name": "test_remove_constraint_pony_pink_gt_2"},
         )
 
+    def test_alter_constraint(self):
+        constraint_1 = models.CheckConstraint(
+            condition=models.Q(pink__gt=2), name="test_alter_constraint_pony_pink_gt_2"
+        )
+        constraint_2 = models.UniqueConstraint(
+            fields=["pink"], name="test_alter_constraint_pony_fields_uq"
+        )
+        project_state = self.set_up_test_model(
+            "test_alterconstraint", constraints=[constraint_1, constraint_2]
+        )
+        ck_operation = migrations.AlterConstraint(
+            "Pony",
+            "test_alter_constraint_pony_pink_gt_2",
+            models.CheckConstraint(
+                condition=models.Q(pink__gt=3),
+                name="test_alter_constraint_pony_pink_gt_2",
+            ),
+        )
+        self.assertEqual(
+            ck_operation.describe(),
+            "Alter constraint test_alter_constraint_pony_pink_gt_2 on Pony",
+        )
+        self.assertEqual(
+            ck_operation.formatted_description(),
+            "~ Alter constraint test_alter_constraint_pony_pink_gt_2 on Pony",
+        )
+        self.assertEqual(
+            ck_operation.migration_name_fragment,
+            "alter_pony_test_alter_constraint_pony_pink_gt_2",
+        )
+        # Test state alteration
```

Comment:
"Test..." is usually avoided in django tests in favor of describing the situation.